internal/namer/namer.go

```diff
@@ -47,4 +53,34 @@ type Namer interface {
 	// zero-based index, or an error if no such name exists.
 	// This method will never be called with a negative index.
 	Fabric(index int) (string, error)
+
+	// Port returns the name of a physical port with the specified parameters,
+	// or an error if no such name exists. This method will never be called with
+	// negative index parameters or an unset or unknown port speed.
```

Comment:
If we guarantee that it will never be called with negative index parameters, would this effectively make this an internal interface that should only be accessed by other packages in this repo so we can make the guarantee? And if so would that mean that we can then make the input parameters unsigned integers?

Reply:
We could ... I've been wondering about this question for Go style. The C++ style guide says "Do not use an unsigned type merely to assert that a variable is non-negative." [1] At one point I thought the Go style asserted the same thing, but either I can't find, they never did, or they removed it.

[1] https://google.github.io/styleguide/cppguide.html#Integer_Types

Reply:
I agree that for a public API, signed integers may make more sense since there is no implicit type conversion in Go and doing explicit conversions is annoying. If this API is unexported though, in spite of this link I think there are two reasons why I think using unsigned integers are more appropriate in this case:
1. I infer that the stated reasons in the C++ guide is because in C++ signed integer overflow/underflow [are undefined](https://stackoverflow.com/questions/10011372/c-underflow-and-overflow) and as a result compilers can do some bug checks and optimizations. Meanwhile for Go this behaviour is [well-defined](https://go.dev/ref/spec#Arithmetic_operators) and as a result these advantages don't apply to Go:
```
For signed integers, the operations +, -, *, /, and << may legally overflow and the resulting value exists and is deterministically defined by the signed integer representation, the operation, and its operands....
```
2. The weird behaviour for unsigned integers happen when you subtract below zero. I don't think this should present an issue because we have decided that the `Namer` implementation would only allow non-negative indices, and as such it should handle these correctly. Essentially the unsigned integer type constrains the input space correctly, and the implementation should be aware of the potential pitfalls for using this type.

Reply:
Based on offline discussion,I'll convert the all of namer to use unsigned ints in a follow-up PR.